Objects/listobject.c

```diff
@@ -151,61 +193,18 @@ _PyList_DebugMallocStats(FILE *out)
 PyObject *
 PyList_New(Py_ssize_t size)
 {
-    PyListObject *op;
-
     if (size < 0) {
         PyErr_BadInternalCall();
         return NULL;
     }
-
-#ifdef WITH_FREELISTS
-    struct _Py_list_freelist *list_freelist = get_list_freelist();
-    if (PyList_MAXFREELIST && list_freelist->numfree > 0) {
-        list_freelist->numfree--;
-        op = list_freelist->items[list_freelist->numfree];
-        OBJECT_STAT_INC(from_freelist);
-        _Py_NewReference((PyObject *)op);
-    }
-    else
-#endif
-    {
-        op = PyObject_GC_New(PyListObject, &PyList_Type);
-        if (op == NULL) {
-            return NULL;
+    PyListObject *op = list_new_prealloc(size);
+    if (op && op->ob_item) {
+        PyObject **items = op->ob_item;
+        for (Py_ssize_t i = 0, n = op->allocated; i < n; i++) {
+            FT_ATOMIC_STORE_PTR_RELEASE(items[i], NULL);
```

Comment:
I'm not sure we need to initialize `ob_item` here at all, but even if we do, we don't need atomic stores.

As written, `list_new_prealloc` already initializes the contents of `ob_item` to NULL. (But I'm not sure that we want `list_new_prealloc` to do that).